yt_dlp/extractor/reddit.py

```diff
@@ -146,6 +148,7 @@ def add_thumbnail(src):
             'thumbnails': thumbnails,
             'timestamp': float_or_none(data.get('created_utc')),
             'uploader': data.get('author'),
+            'channel': data.get('subreddit'),
```

Comment:
`data.get("subreddit")` _already is_ the channel display name ("aww" in this case). Are you talking about the title ("A subreddit for cute and cuddly pictures" - which is not directly available) or the internal id ("t5_2qh1o" or "2qh1o")? The latter comes from `subreddit_id`, and it would probably be consistent to add this as `channel_id` as well.

These are the important values for `test:reddit` (which actually points to [r/videos](https://www.reddit.com/r/videos)

|key|reddit value|yt-dlp field|status|notes|
|---|---|---|---|---|
|author|Antw87|uploader|:white_check_mark:
|author_fullname|t2_15pqie|uploader_id|:watch:|proposal
|created|1501941939.0|-|:no_entry:|use created_utc
|created_utc|1501941939.0|timestamp|:white_check_mark:
|domain|v.redd.it|-|:no_entry:
|downs|0|dislike_count|:white_check_mark:|it's always 0
|id|6rrwyj|id|:white_check_mark:
|name|t3_6rrwyj|-|:no_entry:|same as id
|score|14489|-|:no_entry:|same as like_count
|subreddit|videos|channel|:star:|this patch
|subreddit_id|t5_2qh1e|channel_id|:watch:|proposal
|subreddit_name_prefixed|r/videos|-|:no_entry:|can't be used in filename
|subreddit_subscribers|26633771|-|:no_entry:
|subreddit_type|public|-|:no_entry:
|title|That small heart attack.|title|:white_check_mark:
|ups|14489|like_count|:white_check_mark:|fuzzed
|url|https://v.redd.it/zv89llsvexdz|-|:no_entry:

There are more fields, of course, but most of them aren't really relevant.

Reply:
IMO the "name" of a subreddit would be `r/videos` or `r/aww`

`channel`: `subreddit_name_prefixed`
`channel_id`: `subreddit`
?

Reply:
> IMO the "name" of a subreddit would be `r/videos` or `r/aww`

That's not good for using in filenames, as it contains "/". And it is the same as the subreddit

The patch right now does this and only this:

`channel`: `videos`

Your proposal would give this:

`channel`: `r/videos`  
`channel_id`: `videos`  

My additional proposal would be

`channel`: `videos`  
`channel_id`: `t5_2qh1e`  

But `channel_id` is not as useful.

Reply:
> That's not good for using in filenames, as it contains "/". 

Agreed, but the metadata isn't only for use in filenames. 

> And it is the same as the subreddit

It would indeed be redundant. Yes, my proposal was not good, and I'm not going to argue for it. I was more trying to point out that something like `r/aww` is the practical name of a subreddit.

> Granted, channel_id is not as useful. But it should be more useful than the same thing as channel with a constant prefix.

I think `subreddit_id` (e.g. `t5_2qh1e`) would be equally useless, though. That value seems to be only used internally, and the `subreddit` value is already unique (and permanent).

Reply:
I don't think `subreddit_id` is needed since it is never actually used anywhere in the site and `subreddit` is unique. I was suggesting `channel_id=aww`, `channel=A subreddit for cute and cuddly pictures`. If `channel` can't be extracted, don't worry about it

Reply:
the channel description can be extracted, but it requires a separate API call (to /r/aww/about.json). Maybe in a separate patch...